lib/BackgroundJobs/SubmitContentJob.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+declare(strict_types=1);
+
+namespace OCA\ContextChat\BackgroundJobs;
+
+use OCA\ContextChat\Db\QueueContentItem;
+use OCA\ContextChat\Db\QueueContentItemMapper;
+use OCA\ContextChat\Service\LangRopeService;
+use OCA\ContextChat\Service\ProviderConfigService;
+use OCA\ContextChat\Type\Source;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\BackgroundJob\QueuedJob;
+
+class SubmitContentJob extends QueuedJob {
+	private const BATCH_SIZE = 20;
+
+	public function __construct(
+		ITimeFactory $timeFactory,
+		private LangRopeService $service,
+		private QueueContentItemMapper $mapper,
+		private IJobList $jobList,
+		private ProviderConfigService $configService,
+	) {
+		parent::__construct($timeFactory);
+	}
+
+	/**
+	 * @param $argument
+	 * @return void
+	 */
+	protected function run($argument): void {
+		$entities = $this->mapper->getFromQueue(static::BATCH_SIZE);
+
+		if (empty($entities)) {
+			return;
+		}
+
+		$bucketed = [];
+		foreach ($entities as $entity) {
+			foreach (explode(',', $entity->getUsers()) as $userId) {
+				$bucketed[$userId][] = $entity;
```

Comment:
I believe you will need to initialize the sub-array

Reply:
testing this on php repl worked but sure, can do that.

Reply:
mh. weird, then let's leave it as is :)